src/pytest_memray/marks.py

```diff
@@ -13,25 +18,23 @@
 
 PytestSection = Tuple[str, str]
 
+StackElement = Tuple[str, str, int]
+LeaksFilteringFunction = Callable[[Iterable[StackElement]], bool]
+
 
 @dataclass
 class _MemoryInfo:
-    """Type that holds all memray-related info for a failed test."""
+    """Type that holds memory-related info for a failed test."""
 
     max_memory: float
     total_allocated_memory: int
     allocations: list[AllocationRecord]
     num_stacks: int
     native_stacks: bool
 
-    @property
-    def section(self) -> PytestSection:
-        """Return a tuple in the format expected by section reporters."""
-        total_memory_str = sizeof_fmt(self.total_allocated_memory)
-        max_memory_str = sizeof_fmt(self.max_memory)
+    def _generate_section_text(self, limit_text: str, header_text: str) -> str:
         text_lines = [
-            f"Test is using {total_memory_str} out of limit of {max_memory_str}",
-            "List of allocations: ",
+            f"{header_text} {sizeof_fmt(self.total_allocated_memory)} out of limit of {sizeof_fmt(self.max_memory)}"
```

Comment:
What this prints is very misleading for leaks reports. You get something like:

```text
Test was allowed to leak 5.0KiB per location but at least one location leaked more
----------------------------- memray-leaked-memory -----------------------------
List of leaked allocations: 200.0KiB out of limit of 5.0KiB
    - 100.0KiB allocated here:
        __pyx_pf_6memray_11_test_utils_15MemoryAllocator_16valloc:src/memray/_memray_test_utils.cpp:2905
    - 100.0KiB allocated here:
        __pyx_pf_6memray_11_test_utils_15MemoryAllocator_16valloc:src/memray/_memray_test_utils.cpp:2905
```

This shows two different problems. First, "200.0KiB out of limit of 5.0KiB" is a very weird way to summarize this. 200KiB is a total amount leaked, but 5.0KiB is a per-stack limit, and this wording makes it seem as though they're comparable. I think we should just say "List of leaked allocations:" and leave off everything after the colon, because reporting on the total leaked memory no longer makes sense now that we're deliberately excluding some leaks. And anyway, we already say above "Test was allowed to leak 5.0KiB per location but at least one location leaked more", which does explain the situation pretty well.

Secondly, showing only the final frame of the stacks is totally useless. It'll almost always be `malloc` or `PyObject_Malloc` or something. At best it will tell the user the type of object that was leaked, but not anything more than that. I think we really need to default to showing the full stack here, or at least the most recent handful of frames...

Reply:
Ok, I have added a default of 5 stacks and adapted the code after your suggestions.